src/server/scripts/Spells/spell_mage.cpp

```diff
@@ -341,10 +347,23 @@ class spell_mage_ice_barrier : public SpellScriptLoader
                 }
             }
 
+            void AfterRemove(AuraEffect const* aurEff, AuraEffectHandleModes /*mode*/)
+            {
+                AuraRemoveMode removeMode = GetTargetApplication()->GetRemoveMode();
```

Comment:
oops, i was using it at the start, but figured it was useless checking against removal by enemy spell. i will update it in a moment